shared-libs/validation/src/validation.js

```diff
@@ -178,6 +178,61 @@ module.exports = {
       logger.error('isISOWeek validation failed: the number of week is greater than the maximum');
       return Promise.resolve(false);
     },
+
+    isAfter: (doc, validation) => {
+      const fields = [...validation.funcArgs];
+      try {
+        const duration = _parseDuration(fields.pop());
+        if (!duration.isValid()) {
+          logger.error('isBefore validation: the duration is invalid');
+          return Promise.resolve(false);
+        }
+        const testDate = moment(doc[validation.field]);
+        const controlDate = moment(doc.reported_date).add(duration);
+        if (!testDate.isValid() || !controlDate.isValid()) {
+          logger.error('isBefore validation: date is invalid');
+          return Promise.resolve(false);
+        }
+        if (testDate.isSameOrAfter(controlDate, 'days')) {
+          return Promise.resolve(true);
+        }
+
+        logger.error('isAfter validation failed: the date is older than specified');
+        return Promise.resolve(false);
+
+      }
+      catch (err) {
+        logger.error('isAfter validation: the date or duration is invalid: ' + err.message);
```

Comment:
Please include the whole error message here, for easy debugging: 
```suggestion
        logger.error('isAfter validation: the date or duration is invalid: ' + err.message);
        logger.error('isAfter validation: the date or duration is invalid: %o', err);
```